tests/unit/validation/test_assertions.py

```diff
@@ -28,10 +28,10 @@ def test_validation_exception(errors_to_string_mock: MagicMock):
 def test_assert_valid_input_data(validate_mock: MagicMock):
     validate_mock.return_value = None
     assert_valid_input_data(
-        input_data={"foo": np.array([1])}, calculation_type=CalculationType.state_estimation, symmetric=False
+        input_data={"foo": np.array([1])}, calculation_type=CalculationType.state_estimation, symmetric=False  # type: ignore
```

Comment:
I feel these suppresions might not be needed. And maybe "foo" can be modified to one of the supported type since now we only expect `ComponentTypes` instead of a a generic string.

Reply:
Well you can do 2 things: 1 is test what happens when it actuallily really is forbidden (needs suppression), the other is to remove it entirely because we now actually have the literals, but that means new UB and demanding the user uses a linter/type checker. I'd keep it as is

Reply:
Since the literals don't really enforce type annotations, I believe it should be kept as it is as well.